src/slam_toolbox_common.cpp

```diff
@@ -175,6 +176,7 @@ CallbackReturn SlamToolbox::on_deactivate(const rclcpp_lifecycle::State &)
 /*****************************************************************************/
 {
   RCLCPP_INFO(get_logger(), "Deactivating");
+  stop_threads_.store(true);
```

Comment:
Sorry alsos `stop_threads_.store(false);` in activate so that it can be reactivated.